public/js/repeater.js

```diff
@@ -0,0 +1,185 @@
+window.Repeater = (function() {
+  "use strict"
+
+  var repeatedSets
+
+  // Since we can't depend on Array.prototype.map
+  function map(arr, fn) {
+    var out = new Array(arr.length)
+    arr.forEach(function(x, i) { out[i] = fn(x, i) })
+    return out
+  }
+
+  function indexBy(key, arr) {
+    var out = {}
+    arr.forEach(function(x) { out[x[key]] = x })
+    return out
+  }
+
+  function Block() { this.init.apply(this, arguments) }
+  (function(_) {
+    _.init = function(el) {
+      var self = this // closure fix
+
+      this.name = el.id
+      this.container = el
+
+      this.instances = map(el.querySelectorAll('.repeater-instance'), function(fieldset, i) {
+        var instance = new Instance(self, fieldset, i)
+        instance.reindex(i)
+        return instance
+      })
+    }
+
+    _.setupListeners = function() {
+      // we use one global listener for removal, because the repeat links may
+      // get added and removed from the DOM, and this means we don't have to
+      // re-register the event listeners when we clone the nodes.
+      this.container.addEventListener('click', function(evt) {
+        if (!evt.target.classList.contains('remove-repeat-link')) return
+
+        evt.preventDefault()
+        var instance = instanceFor(evt.target)
+        instance && instance.remove()
+      })
+
+      return this
+    }
+
+    _.repeat = function() {
+      if (!this.instances.length) throw new Error('empty instances, can\'t repeat!')
```

Comment:
Would prefer to use template literals vs escaping stings `empty instances, can't repeat!`